<launch>

    <arg name="marker_size" doc="Size of the ArUco marker used, in meters"/>
    <arg name="marker_id" doc="The ID of the ArUco marker used"/>

    <!-- start camera driver -->
    <include file="$(find roboticvisiondataset)/launch/rs_l515.launch" >
    </include>

    <!-- start ArUco -->
    <include file="$(find roboticvisiondataset)/launch/aruco.launch" >
        <arg name="camera" value="/rs_l515/color" />
        <arg name="frame" value="rs_l515_color_optical_frame" />
    </include>

    <!-- start the robot -->
    <include file="$(find iiwa_moveit)/launch/moveit_planning_execution.launch">
        <arg name="sim" value="false" />
        <arg name="rviz" value="true" />
    </include>

    <!-- start easy_handeye -->
    <node pkg="roboticvisiondataset" type="handeye_calibration.py" name="handeye_calibration" args="--force-discover" output="screen">
        <param name="eye_on_hand" value="true" />
        <param name="start_rviz" value="false"/>
        <param name="move_group_namespace" value="/iiwa" />
        <param name="move_group" value="manipulator" />
        <param name="rviz_config_file" value="$(find easy_handeye)/launch/iiwa_stack_config.rviz" />
        <param name="tracking_base_frame" value="rs_l515_color_optical_frame" />
        <param name="tracking_marker_frame" value="camera_marker" />
        <param name="robot_base_frame" value="iiwa_link_0" />
        <param name="robot_effector_frame" value="iiwa_link_ee" />
        <param name="freehand_robot_movement" value="false" />
        <param name="robot_velocity_scaling" value="0.3" />
        <param name="robot_acceleration_scaling" value="0.1" />
        <param name="angle_delta" value="10" />
        <param name="translation_delta" value="0.1" />
        <param name="camera_topic" value="/rs_l515/color" />
    </node>

</launch>
